drivers/cyw43/cywbt.c

```diff
@@ -29,42 +29,48 @@
 
 #include "py/runtime.h"
 #include "py/mphal.h"
-#include "pin_static_af.h"
-#include "uart.h"
 #include "extmod/mpbthci.h"
 
 #if MICROPY_PY_NETWORK_CYW43
 
 #include "lib/cyw43-driver/src/cyw43_config.h"
 #include "lib/cyw43-driver/firmware/cyw43_btfw_4343A1.h"
 
-// Provided by the port.
-extern pyb_uart_obj_t mp_bluetooth_hci_uart_obj;
-
 // Provided by the port, and also possibly shared with the stack.
 extern uint8_t mp_bluetooth_hci_cmd_buf[4 + 256];
 
 /******************************************************************************/
 // CYW BT HCI low-level driver
 
+#ifdef CYW43_PIN_BT_CTS
+// This code is not portable and currently only builds on stm32 port.
+
+#include "pin_static_af.h"
+#include "uart.h"
+
+// Provided by the port.
+extern pyb_uart_obj_t mp_bluetooth_hci_uart_obj;
+
 STATIC void cywbt_wait_cts_low(void) {
-    mp_hal_pin_config(pyb_pin_BT_CTS, MP_HAL_PIN_MODE_INPUT, MP_HAL_PIN_PULL_UP, 0);
+    mp_hal_pin_config(CYW43_PIN_BT_CTS, MP_HAL_PIN_MODE_INPUT, MP_HAL_PIN_PULL_UP, MP_HAL_PIN_AF_GPIO);
```

Comment:
Passing 0 in the last argument is simply because it's an unused argument.  There is no such thing as a GPIO alt function.

So, either keep it as 0, or have a constant named `MP_HAL_PIN_AF_NONE`.

Reply:
The `alt` argument is used in `mp_hal_pin_config` in `stm32` port, and I defined `MP_HAL_PIN_AF_GPIO` as 0, for `mimxrt` the GPIO alt function is `5`

Reply:
But the alt argument is only needed on stm32 if MP_HAL_PIN_MODE_ALT is used.  And it should be this way on other ports as well.  If mimxrt needs to set alt to 5 then it should do that internally in the function when MP_HAL_PIN_MODE_INPUT/OUTPUT is passed in as the mode.

Really the alt argument should be removed altogether from this function, and functions like mp_hal_pin_config_alt used for non-GPIO cases.

Reply:
> . If mimxrt needs to set alt to 5 then it should do that internally in the function

This is possible, but I was trying to avoid it by just defining the GPIO AF value to 0 or 5 (for mimxrt), but I will revert this to `0` and handle it in mimxrt. Note switching to GPIO to read the pin is not actually needed, at least for the ports that currently support this driver.

Reply:
I reverted `MP_HAL_PIN_AF_GPIO` to 0.

> Really the alt argument should be removed altogether from this function, and functions like mp_hal_pin_config_alt used for non-GPIO cases.

`mp_hal_pin_config_alt` only exists to support the `stm32` instance number (for example `AF_FN_UART, 1`), other ports just use `mp_hal_pin_config`. Anyway, `cywbt_wait_cts_low` is not portable, because it needs `AF_FN_UART`, I think it can be made portable if the pin mode is not changed, or if this function is made abstract and implemented by ports (for example `mp_bluetooth_hci_uart_poll_cts`).